calibre: cssutils -> css-parser

jethrokuan opened the pull request:
###### Motivation for this change

Updates an altered calibre dependency. Calibre now fails with:

```
ImportError: No module named css_parser.css
```

because of a recent change https://github.com/kovidgoyal/calibre/commit/dd7d8ea3c4f47e315623d882d6b68339da959ab9

This also introduces a new package css-parser, which calibre uses.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

dotlambda reviewed (changes requested):
Adding the python package should be a separate commit.

jethrokuan commented on pkgs/development/python-modules/css-parser/default.nix:
```diff
@@ -0,0 +1,24 @@
+    sha256 = "c7ab355512ae51334ba6791a7e4d553f87bef17ba2026f1cc9bf3b17a7779d44";
+  };
+
+  buildInputs = [ mock ];
```
These were all inherited from the previous package `cssutils`. `css-parser` is just an updated fork of `cssutils` with minor changes, so I assumed nothing much had changed.

dotlambda replied:
I.e. you're not sure whether it belongs there. Please remove it and remember to enable tests (and possibly add `checkInputs`) once a new version is released.

jethrokuan replied:
Looks like any dependency on it has been removed. Sorry for that, should've done the due diligence.

jethrokuan commented:
Taking your reviews into account, but closing this PR to split it into 2.

dotlambda commented:
No need for two PRs, just two commits.

jethrokuan committed:
pythonPackages.css-parser: init at 1.0.4

jethrokuan committed:
calibre: fix altered dependency

Calibre has changed its dependency from cssutils to css-parser in
https://github.com/kovidgoyal/calibre/commit/dd7d8ea3c4f47e315623d882d6b68339da959ab9

jethrokuan commented:
done, thanks for reviewing again @dotlambda

dotlambda commented on pkgs/development/python-modules/css-parser/default.nix:
```diff
@@ -0,0 +1,22 @@
+{ lib, buildPythonPackage, fetchPypi, mock }:
```
```suggestion
{ lib, buildPythonPackage, fetchPypi }:
```

dotlambda commented:
@GrahamcOfBorg build python2.pkgs.css-parser python3.pkgs.css-parser

dotlambda commented:
@GrahamcOfBorg build calibre
